hazelcast/src/main/java/com/hazelcast/map/impl/tx/TransactionalMapProxy.java

```diff
@@ -315,12 +319,9 @@ public void delete(Object key) {
     @SuppressWarnings("unchecked")
     public Set keySet(Predicate predicate) {
         checkTransactionState();
-        if (predicate == null) {
-            throw new NullPointerException("Predicate should not be null!");
-        }
-        if (predicate instanceof PagingPredicate) {
-            throw new NullPointerException("Paging is not supported for Transactional queries!");
-        }
+        checkNotNull(predicate, "Predicate should not be null!");
+        checkNotInstanceOf(PagingPredicate.class, predicate, "Paging is not supported for Transactional queries!");
```

Comment:
I think the original code can be seen as a bug since the throw exception is not correct. So a bug fix would be to make it an IllegalArgumentException